src/domain/friendship_event_validator.rs

```diff
@@ -3,8 +3,18 @@ use crate::{
     entities::friendship_history::FriendshipHistory,
 };
 
-/// Validates the new event is valid and different from the last recorded.
+// Validates that the transition between current state and new event is valid and the acting user is authorized to perform the action
```

Comment:
I used three `///` because otherwise, when u hover over the func where it is called, u won't get the documented info, check it out [here](https://doc.rust-lang.org/reference/comments.html)